telegram/ext/_extbot.py

```diff
@@ -546,29 +552,40 @@ def _effective_inline_results(
         return results, next_offset
 
     @no_type_check  # mypy doesn't play too well with hasattr
-    def _insert_defaults_for_ilq_results(self, res: "InlineQueryResult") -> None:
+    def _insert_defaults_for_ilq_results(self, res: "InlineQueryResult") -> "InlineQueryResult":
         """This method is called by Bot.answer_inline_query to replace `DefaultValue(obj)` with
         `obj`.
         Overriding this to call insert the actual desired default values.
         """
+        # Copy the objects that need modification to avoid modifying the original object
+        copied = False
         if hasattr(res, "parse_mode") and res.parse_mode is DEFAULT_NONE:
+            res = copy(res)
+            copied = True
             res.parse_mode = self.defaults.parse_mode if self.defaults else None
         if hasattr(res, "input_message_content") and res.input_message_content:
             if (
                 hasattr(res.input_message_content, "parse_mode")
                 and res.input_message_content.parse_mode is DEFAULT_NONE
             ):
+                if not copied:
+                    res = copy(res)
+                    copied = True
                 res.input_message_content.parse_mode = (
                     self.defaults.parse_mode if self.defaults else None
                 )
             if (
                 hasattr(res.input_message_content, "disable_web_page_preview")
                 and res.input_message_content.disable_web_page_preview is DEFAULT_NONE
             ):
+                if not copied:
+                    res = copy(res)
```

Comment:
uff … I'll have to introduce a subclass of `InputMessageContent` to the tests that has `disable_web_page_preview` bot not `parse_mode`, then. The things we do to prevent regressions xD

Reply:
I wonder if our test suite is larger than the core code base itself..

Reply:
Note yet, but it's close :D 

```python
import pathlib

tg_loc = 0
test_loc = 0

for file in pathlib.Path("telegram").rglob("*.py"):
    tg_loc += len(file.read_text("utf-8").splitlines())

for file in pathlib.Path("tests").rglob("*.py"):
    test_loc += len(file.read_text("utf-8").splitlines())

print(f"Telegram: {tg_loc} lines")
print(f"Tests: {test_loc} lines")
```

> Telegram: 49476 lines
> Tests: 41555 lines